### ISSUE TITLE
Migration unsuccessful with transformEntries because "Cannot edit archived" and "Cannot publish archived"

### ISSUE THREAD

**Issue by nicco91** [contentful/contentful-migration#122]
I'm trying to do a migration which create a fields and then transform entries creating values for the new field created before.

## Expected Behavior

Run the migration, create the field, fill it with the values I create.

## Actual Behavior

Field is created correctly, values are generated correctly, but the migration fails because it try to publish an entry which is in status `archived`.

```
$ node migrate.js --portal tcom --env test --migration migrations/3.13.0.js
The following migration has been planned

Update Content Type pack

  Create field servicesDescription
    - name: "Services Description"
    - type: "Text"

Publish Content Type pack
Transform entries for pack
  - from: services
  - to: servicesDescription
? Do you want to apply the migration Yes
 √ Update Content Type pack
 > Transform entries for pack
   × Making requests (48/48)
     → Batch failed
🚨  Migration unsuccessful
Batch failed

Error: {"status":"Bad Request","message":"Cannot edit archived","details":{},"url":"https://api.contentful.com:443/spaces/fzchbmaavok4\\environments\\develop\\entries\\2iqxnK1F1aakq24U4C8CSq"}


Error: {"status":"Bad Request","message":"Cannot publish archived","details":{},"url":"https://api.contentful.com:443/spaces/fzchbmaavok4\\environments\\develop\\entries\\2iqxnK1F1aakq24U4C8CSq\\published"}
```

I have no control on the entries status with the `transformEntries` migration function. I just provide the `transformEntryForLocale` function.
So, how can I run this migration skipping the archived entries?
It actually works for the entries that are not `archived` but the error message is shown and it says "migration unsuccessful", even if it is just partially unsuccessful.

## Possible Solution

Parameters on the `transformEntries` function for choose entries status or just skip the archived entries and return a success message. 

## Steps to Reproduce

Run any migration with `transformEntries` function where there are some `archived` entries.

## Environment

-   **Node Version**: 8.11.3
-   **Package Manager Version**: 5.6.0
-   **Operating System**: Windows 10 64-bit
-   **Package Version**: 0.10.0

**Comment by stoltzrobin** [comment]
@Khaledgarbaya have there been any progress on this?

**Comment by Khaledgarbaya** [comment]
This should be fixed by https://github.com/contentful/contentful-migration/pull/143

**Comment by TimBeyer** [comment]
I think this will need a separate fix from #143 
I'm not sure if there is a use case for deriving things from archived entries, but I could imagine there being one. The very simplest way would be to simply filter out any archived entities during fetching.
However, if anyone would have a use case for deriving from archived entities as well, we'll have to make it a bit more complicated than that.

Probably the best in that case would be to filter this in the transform/derive action itself where `getEntriesForContentType` gets called since it'll be quite a bit more complicated to change the fetching code to be aware of when archived entries might be needed and when not.

Then we will need a new config option for `includeArchived`, but only for the derive type of actions.

I think I'd prefer if we could simply skip archived entities altogether, and only introduce this as an explicit option later in case someone really needs to derive something from archived entities without moving them back into a draft state.
